[Python][Flaky Test] test_sync_pubsub_sharded_max_size_message_callback

### Test Name

tests/sync_tests/test_sync_pubsub.py::TestSyncPubSub::test_sync_pubsub_sharded_max_size_message_callback[

### System Information

x86_64-unknown-linux-gnu

### Language and Version

Python 3.9

### Engine Version

7.0, 7.2, 8.0, 8.1

### Logs

```
FAILED tests/sync_tests/test_sync_pubsub.py::TestSyncPubSub::test_sync_pubsub_sharded_max_size_message_callback[SubscriptionMethod.Config-True] - assert 0 == 1
 +  where 0 = len([])
FAILED tests/sync_tests/test_sync_pubsub.py::TestSyncPubSub::test_sync_pubsub_sharded_max_size_message_callback[SubscriptionMethod.Lazy-True] - assert 0 == 1
 +  where 0 = len([])
FAILED tests/sync_tests/test_sync_pubsub.py::TestSyncPubSub::test_sync_pubsub_sharded_max_size_message_callback[SubscriptionMethod.Blocking-True] - assert 0 == 1
 +  where 0 = len([])
```

### Screenshots

_No response_

### Glide Version

main

### Potential Fix

Test failures occurred after the merging of [Go, Java, Python: Clean-up pubsub in wrappers and expand coverage
#5468 ](https://github.com/valkey-io/valkey-glide/pull/5468/changes)

--- a/python/tests/async_tests/test_pubsub.py
+++ b/python/tests/async_tests/test_pubsub.py
@@ -38,6 +38,7 @@ from tests.utils.pubsub_test_utils import (
     subscribe_by_method,
     sunsubscribe_by_method,
     unsubscribe_by_method,
+    wait_for_messages,
     wait_for_subscription_state,
     wait_for_subscription_state_if_needed,
 )
@@ -2271,10 +2272,6 @@ class TestPubSub:
             await pubsub_client_cleanup(publishing_client)
 
     @pytest.mark.skip_if_version_below("7.0.0")
-    @pytest.mark.skip(
-        reason="This test requires special configuration for client-output-buffer-limit for valkey-server and timeouts seems "
-        + "to vary across platforms and server versions"
-    )
     @pytest.mark.parametrize("cluster_mode", [True])
     @pytest.mark.parametrize(
         "subscription_method",
@@ -2284,26 +2281,29 @@ class TestPubSub:
             SubscriptionMethod.Blocking,
         ],
     )
-    async def test_pubsub_sharded_max_size_message_callback(
+    async def test_pubsub_sharded_large_size_message_callback(
         self, request, cluster_mode: bool, subscription_method: SubscriptionMethod
     ):
         """
-        Tests publishing and receiving maximum size messages in sharded PUBSUB with callback method.
+        Tests publishing and receiving large messages in sharded PUBSUB with callback method.
 
-        This test verifies that very large messages (512MB - BulkString max size) can be published and received
+        This test verifies that large messages can be published and received
         correctly. It ensures that the PUBSUB system
-        can handle maximum size messages without errors and that the callback message
+        can handle large messages without errors and that the callback message
         retrieval method works as expected.
 
         The test covers the following scenarios:
         - Setting up PUBSUB subscription for a specific sharded channel with a callback.
-        - Publishing a maximum size message to the channel.
+        - Publishing a large message to the channel.
         - Verifying that the message is received correctly using the callback method.
         """
         publishing_client, listening_client = None, None
         try:
-            channel = "sharded_max_size_callback_channel"
-            message = "0" * 512 * 1024 * 1024
+            channel = "sharded_large_size_callback_channel"
+            # 12MB: large enough to exercise the large-message path, but below the
+            # default client-output-buffer-limit hard cap (32MB) so the server won't
+            # disconnect the subscriber.
+            message = "0" * 12 * 1024 * 1024
 
             callback_messages: List[PubSubMsg] = []
             callback, context = new_message, callback_messages
@@ -2337,6 +2337,7 @@ class TestPubSub:
                 listening_client,
                 subscription_method,
                 expected_sharded={channel},
+                timeout_ms=10000,
             )
 
             assert (
@@ -2346,8 +2347,8 @@ class TestPubSub:
                 == 1
             )
 
-            # allow the message to propagate
-            await anyio.sleep(15)
+            # Wait for message with polling
+            wait_for_messages(1, callback_messages, timeout=45.0)
 
             assert len(callback_messages) == 1
 

--- a/python/tests/async_tests/test_pubsub.py
+++ b/python/tests/async_tests/test_pubsub.py
@@ -38,7 +38,6 @@ from tests.utils.pubsub_test_utils import (
     subscribe_by_method,
     sunsubscribe_by_method,
     unsubscribe_by_method,
-    wait_for_messages,
     wait_for_subscription_state,
     wait_for_subscription_state_if_needed,
 )
@@ -2347,8 +2346,8 @@ class TestPubSub:
                 == 1
             )
 
-            # Wait for message with polling
-            wait_for_messages(1, callback_messages, timeout=45.0)
+            # Wait for message to propagate with longer timeout
+            await anyio.sleep(30)
 
             assert len(callback_messages) == 1
 

--- a/python/tests/sync_tests/test_sync_pubsub.py
+++ b/python/tests/sync_tests/test_sync_pubsub.py
@@ -1926,10 +1926,6 @@ class TestSyncPubSub:
             assert callback_messages[0].channel == channel.encode()
             assert callback_messages[0].pattern is None
 
-    @pytest.mark.skip(
-        reason="This test requires special configuration for client-output-buffer-limit for valkey-server and timeouts seems "
-        + "to vary across platforms and server versions"
-    )
     @pytest.mark.skip_if_version_below("7.0.0")
     @pytest.mark.parametrize("cluster_mode", [True])
     @pytest.mark.parametrize(
@@ -1940,25 +1936,28 @@ class TestSyncPubSub:
             SubscriptionMethod.Blocking,
         ],
     )
-    def test_sync_pubsub_sharded_max_size_message_callback(
+    def test_sync_pubsub_sharded_large_size_message_callback(
         self, request, cluster_mode: bool, subscription_method: SubscriptionMethod
     ):
         """
-        Tests publishing and receiving maximum size messages in sharded PUBSUB with callback method.
+        Tests publishing and receiving large messages in sharded PUBSUB with callback method.
 
-        This test verifies that very large messages (512MB - BulkString max size) can be published and received
+        This test verifies that large messages can be published and received
         correctly. It ensures that the PUBSUB system
-        can handle maximum size messages without errors and that the callback message
+        can handle large messages without errors and that the callback message
         retrieval method works as expected.
 
         The test covers the following scenarios:
         - Setting up PUBSUB subscription for a specific sharded channel with a callback.
-        - Publishing a maximum size message to the channel.
+        - Publishing a large message to the channel.
         - Verifying that the message is received correctly using the callback method.
         """
 
         channel = get_random_string(10)
-        message = "0" * 512 * 1024 * 1024
+        # 12MB: large enough to exercise the large-message path, but below the
+        # default client-output-buffer-limit hard cap (32MB) so the server won't
+        # disconnect the subscriber.
+        message = "0" * 12 * 1024 * 1024
 
         callback_messages: List[PubSubMsg] = []
         callback, context = new_message, callback_messages
@@ -1977,6 +1976,7 @@ class TestSyncPubSub:
                 listening_client,
                 subscription_method,
                 expected_sharded={channel},
+                timeout_sec=10.0,
             )
 
             assert (

